app/controllers/aid_applications/applicants_controller.rb

```diff
@@ -24,6 +24,16 @@ def update
       elsif params[:form_action] == 'allow_mailing_address'
         @aid_application.allow_mailing_address = true
         @aid_application.save
+      elsif params[:form_action] == 'remove_mailing_address'
+        @aid_application.allow_mailing_address = false
+        @aid_application.update_attributes({
+                                             mailing_street_address: nil,
+                                             mailing_apartment_number: nil,
+                                             mailing_city: nil,
+                                             mailing_state: nil,
+                                             mailing_zip_code: nil
+                                           })
+        @aid_application.save
```

Comment:
fyi, that `update_attributes` actually saves things in the database. I think that might just be a typo for `assign_attributes`: https://davidverhasselt.com/set-attributes-in-activerecord/

I think this could all be collapsed into one line. Also, the reason for 99% of the `assign_attributes / save` pattern is because `update` does not accept a `context` argument.

```suggestion
      elsif params[:form_action] == 'remove_mailing_address'
        @aid_application.update(
          allow_mailing_address: false
          mailing_street_address: nil,
          mailing_apartment_number: nil,
          mailing_city: nil,
          mailing_state: nil,
          mailing_zip_code: nil
)
```